### PR TITLE
feat: add leaf spec

### PR DESCRIPTION
**Pull request by neo773**
This PR adds auto complete spec for [Leaf PHP](https://leafphp.dev/) CLI

https://user-images.githubusercontent.com/62795688/219960742-3e75dc91-4cd4-4708-b777-7adf61bdc722.mp4

### CODE REVIEW

**Comment by neo773** [comment]
I have read the CLA Document and I hereby sign the CLA

**Comment by neo773** [comment]
recheck

File: src/leaf.ts [hunk 1]
```diff
@@ -0,0 +1,120 @@
+const completionSpec: Fig.Spec = {
+  name: "leaf",
+  description: "",
```
Comment (fedeci) [inline src/leaf.ts:3]:
```suggestion
  description: "Create and interact with your leaf projects",
```

File: src/leaf.ts [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+      description: "Do not output any message",
+    },
+    {
+      name: ["-V", "--version"],
+      description: "Display this application version",
+    },
+    {
+      name: ["--ansi", "--no-ansi"],
+      description: "Force (or disable --no-ansi) ANSI output",
+    },
```
Comment (fedeci) [inline src/leaf.ts:21]: These are two different options, one opposite of the other and must be listed separately:
```suggestion
    {
      name: "--ansi",
      description: "Force ANSI output",
      exclusiveOn: ["--no-ansi"],
    },
    {
      name: "--no-ansi",
      description: "Disable ANSI output",
      exclusiveOn: ["--ansi"],
    },
```

File: src/leaf.ts [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+      name: "list",
+      description: "List commands",
+    },
+    {
+      name: "run",
+      description: "Run a script in your composer.json",
+      args: {
+        name: "command name",
+        description: "Name of the command",
+      },
```
Comment (fedeci) [inline src/leaf.ts:75]: If you want this may be improved with a custom generator.

File: src/leaf.ts [hunk 4]
```diff
@@ -0,0 +1,120 @@
...
+    {
+      name: "test",
+      description: "Test your leaf application through leaf alchemy",
+      subcommands: [
+        {
+          name: "test:setup",
+          description: "Add tests to your application",
+        },
+      ],
+    },
```
Comment (fedeci) [inline src/leaf.ts:104]: This gets run with the following command, is it intended? `leaf test test:setup`

**Review by fedeci — CHANGES_REQUESTED** [review]
Thanks @neo773! I left some comments, I'll wait for a feedback!

**Comment by fedeci** [comment]
@neo773 I am reopening and committing this one, thanks for the work!

**Comment by fedeci** [comment]
See [3604f26](https://github.com/withfig/autocomplete/commit/3604f26df60b2a3d800704bafdbbc96710833be0)

**Pull request closed without merging**